### PR TITLE
Pilotmover preparator and small fixes in stageInTest

### PR DESCRIPTION
**Pull request by DanilaOleynik**

### CODE REVIEW

**Comment by tmaeno** [comment]
What is the difference from rse_preparator?

**Comment by DanilaOleynik** [comment]
Only other names, and comments. Basically, Rucio does not make transfer in case of usage of Pilot mover.  Rucio return used to find proper replica for transfer, but transfer itself goes through gfal warped in rucio client. So, there is no local RSE, and local copy of file does not registered in Rucio.

**Comment by dougbenjamin** [comment]
Is there a reason not to use rse_preparator?

**Comment by DanilaOleynik** [comment]
I don't get the question. rse_praparator should interface with Rucio to trigger the transfer, pilot mover does not work in this way: it perform transfer itself.

**Comment by dougbenjamin** [comment]
Hi Danilla,

    Since pilots are not running at HPC's in harvester and at least two HPC's do not have local RSE's onsite, we must 
use rucio client to
fetch data.

Doug

**Comment by DanilaOleynik** [comment]
Hi Doug,

We are not going to launch pilots, we are used API for some Pilot functionality, in these cases to Pilot movers. I other words: we deploy Pilot as set of libraries and Harvester perform calls to these libraries. 
We will have more overlapping with Pilot for: experiment specific environment preparation, analyzing of results of execution of payloads etc. 

Cheers,
Danila

**Comment by tmaeno** [comment]
The preparator fetches data from remote RSEs using the data mover component of the pilot.
So both "rse_" and "pilotmover_" are somehow relevant for the name of the preparator. Note that
if there are local RSEs panda instead of harvester accesses to rucio to transfer files to the local
SEs, so there will not be a preparator to directly interface with rucio and thus the name of "rse_"
doesn't have to be reserved for that. I don't really care about class names unless they are extremely
confusing, but I don't want to have two files for the same usecase in the repo. Can you do

$ git mv rse_praparator.py pilotmover_preparator.py

and add comments/changes to it, rather than adding the new file?

**Comment by DanilaOleynik** [comment]
Hi Tadashi,

Please, check recent changes. 

Cheers,
Danila

**Comment by tmaeno** [comment]
Is basePath a static config parameter? Or is it somehow dynamically changed by using setBasePath() during harvester is running?
BTW, I see that stageInTest.py uses setBasePath(), but what calls the method in the real operation and when?

**Comment by dougbenjamin** [comment]
Inside queue configuration of ALCF_Theta there is this section -
	       "preparator":{
			"name":"RsePreparator",
			"module":"pandaharvester.harvesterpreparator.rse_preparator",
			"basePath":"/projects/AtlasADSP/atlas/harvester/rucio"
			},

This defines the storage area where the (at least the top of the storage tree) where input and outfiles
should be storage at an HPC site.

line #19 of the pull request -

  __basePath = os.getcwd()

This ensures that files are placed under where harvester is running.   At ALCF and NERSC that 
is not necessarily the same location. At NERSC we want to store files in scratch space but save
code if possible in project space.    This is why a static area was defined in the local queue
configuration.

File: pandaharvester/harvesterpreparator/pilotmover_preparator.py [hunk 1]
```diff
@@ -70,13 +69,18 @@ def trigger_preparation(self, jobspec):
...
+    #    return self.__basePath
+
     # resolve input file paths
     def resolve_input_paths(self, jobspec):
         # get input files
         inFiles = jobspec.get_input_file_attributes()
         # set path to each file
         for inLFN, inFile in inFiles.iteritems():
-            inFile['path'] = mover_utils.construct_file_path(self.basePath, inFile['dataset'], inFile['scope'], inLFN)
+            inFile['path'] = self.basePath
```
Comment (tmaeno) [inline pandaharvester/harvesterpreparator/pilotmover_preparator.py]: This line should set the full path to the file instead of the base dir name.

Comment (dougbenjamin) [inline pandaharvester/harvesterpreparator/pilotmover_preparator.py]: The base dir name only is part of the top of the storage tree. construct_file_path creates the file location account the rucio path convention. This is why the original code used the construct_file_path.

**Comment by DanilaOleynik** [comment]
Hi Doug,

Why we need rucio path convention on local storage without rucio? 

Cheers,
Danila

**Comment by dougbenjamin** [comment]
it avoids overloading directories with too many files.

**Comment by DanilaOleynik** [comment]
I will push the changes in few minutes. 
With creating of directory tree according to Rucio convention, we should think - how Harvester will cleanup these directories in safe way.

**Pull request merged**